### PR TITLE
Check off most of the graduation checklist for casemap

### PR DESCRIPTION
**Pull request by Manishearth**
Progress on https://github.com/unicode-org/icu4x/issues/3234

### CODE REVIEW

File: experimental/casemap/Cargo.toml [hunk 1]
```diff
@@ -47,17 +45,22 @@ icu_casemap_data = { version = "~1.3.0", path = "data", optional = true }
 
 [features]
 default = ["compiled_data"]
-std = ["icu_collections/std", "icu_provider/std"]
```
Comment (robertbastian) [inline experimental/casemap/Cargo.toml:50]: I believe other crates have an `std` feature even if it doesn't do anything. We might need it in the future.

**Review by robertbastian — APPROVED** [review]

File: experimental/casemap/benches/casemap.rs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    let root = langid!("und");
+    let tr = langid!("tr");
+    c.bench_function("icu_casemap/overview", |b| {
+        b.iter(|| {
+            black_box(casemapper.lowercase_to_string(black_box(TEST_STRING_EN), &root));
+            black_box(casemapper.uppercase_to_string(black_box(TEST_STRING_EN), &root));
+        });
+    });
+
+    c.bench_function("icu_casemap/titlecase_segment", |b| {
```
Comment (sffc) [inline experimental/casemap/benches/casemap.rs:28]: Nit: Gate non-core benches behind the "bench" feature

File: experimental/casemap/src/casemapper.rs [hunk 3]
```diff
@@ -14,6 +14,18 @@ use writeable::Writeable;
 /// A struct with the ability to convert characters and strings to uppercase or lowercase,
 /// or fold them to a normalized form for case-insensitive comparison.
 ///
+/// # Example
```
Comment (sffc) [inline experimental/casemap/src/casemapper.rs:17]: Nit: supposed to be "Examples"

**Review by sffc — COMMENTED** [review]
(requesting changes)

**Comment by Manishearth** [comment]
https://github.com/unicode-org/icu4x/pull/3708

**Pull request merged**